lib/chef/provider/apt_update.rb

```diff
@@ -75,6 +77,13 @@ def do_update
         declare_resource(:execute, "apt-get -q update")
       end
 
+      def self.uses_apt?
+        ENV["PATH"] ||= ""
+        paths = %w{ /bin /usr/bin /sbin /usr/sbin } + ENV["PATH"].split(::File::PATH_SEPARATOR)
+        paths.any? { |path| ::File.executable?(::File.join(path, "apt-get")) }
```

Comment:
I'm pretty sure i tried to use the which helper and it didn't work, but i'm hazy on the details right now. And yes, eventually we should extract the `uses_apt?` method.